### PR TITLE
Create example flag qss

### PR DESCRIPTION
**Pull request by knsmith**
Solves issue https://github.com/SupertechLabs/general-superstaq/issues/158 for qss.

Added example-requirements.txt and edited setup.py

### CODE REVIEW

**Review by pgoiporia — COMMENTED** [review]
LGTM! Have you tested this out in a dummy environment? If so, I'll approve.

**Comment by knsmith** [comment]
I tested with / without the examples flag to confirm that packages were / were not present, and it works! I needed to update general-superstaq requirements (check failed originally), but it seems to be fixed now!

**Review by pgoiporia — DISMISSED** [review]

File: setup.py [hunk 1]
```diff
@@ -37,7 +41,7 @@
     author_email="pranav@super.tech",
     python_requires=(">=3.7.0"),
     install_requires=requirements,
-    extras_require={"dev": dev_requirements},
+    extras_require={"dev": dev_requirements, "example": example_requirements},
```
Comment (richrines1) [inline setup.py:44]: nit
```suggestion
    extras_require={"dev": dev_requirements, "examples": example_requirements},
```

Comment (vtomole) [reply setup.py:44]: https://github.com/SupertechLabs/cirq-superstaq/pull/358#discussion_r995160320

**Comment by knsmith** [comment]
@pgoiporia I just corrected the doc string. Should be ready for review!

**Review by pgoiporia — APPROVED** [review]
LGTM. Nice!

**Comment by knsmith** [comment]
Hi @pgoiporia I'm so sorry could you review this again? Thank you!

**Pull request merged**